augur/filter.py

```diff
@@ -362,6 +406,33 @@ def filter_by_non_nucleotide(metadata, sequence_index):
     return set(filtered_sequence_index[no_invalid_nucleotides].index.values)
 
 
+def include(metadata, include_files):
+    """Include strains in the given list of text files from the given metadata.
+
+    Parameters
+    ----------
+    metadata : pandas.DataFrame
+        Metadata indexed by strain name
+    include_files : list[str]
+        List of filenames with strain names to include from the given metadata
+
+    Returns
+    -------
+    set[str]:
+        Strains that pass the filter
+
+    >>> metadata = pd.DataFrame([{"region": "Africa"}, {"region": "Europe"}], index=["strain1", "strain2"])
+    >>> with NamedTemporaryFile(delete=False) as include_file:
+    ...     characters_written = include_file.write(b'strain1')
+    >>> include(metadata, [include_file.name])
+    {'strain1'}
+    >>> os.unlink(include_file.name)
+
+    """
+    included_strains = read_strains(*include_files)
+    return set(metadata.index.values) & included_strains
+
+
 def include_by_query(metadata, include_where):
```

Comment:
We should probably rename `include_by_query` to `include_by_include_where` for consistency with other functions. But `include_by_include_where` is a silly name.

Reply:
Silly yes, but I still concur with "should rename".

Or the third option: systematically rename to something else that is both less silly but still sensical and consistent. `include_by_query(metadata, include_where)` could be `include_where(metadata, condition)`, for example.

Reply:
I like the third option! I'll add this to the checklist of [the latest PR](https://github.com/nextstrain/augur/pull/750).